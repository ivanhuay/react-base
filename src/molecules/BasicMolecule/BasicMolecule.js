import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import * as styles from './BasicMolecule.module.scss';
import BasicAtom from '../../atoms/BasicAtom';

const cx = classNames.bind(styles);

const BasicMolecule = ({ title }) => (
  <div className={cx('container')}>
    <BasicAtom title={title} />
  </div>
);

BasicMolecule.defaultProps = {
  title: '',
};

BasicMolecule.propTypes = {
  title: PropTypes.string,
};
export default BasicMolecule;
