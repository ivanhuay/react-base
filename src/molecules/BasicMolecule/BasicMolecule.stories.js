import React from 'react';

import BasicMolecule from './BasicMolecule';

export default {
  component: BasicMolecule,
  title: 'molecules/BasicMolecule',
};

export const Default = () => <BasicMolecule>default</BasicMolecule>;
