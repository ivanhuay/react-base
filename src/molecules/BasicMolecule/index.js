import BasicMolecule from './BasicMolecule';

export default BasicMolecule;
