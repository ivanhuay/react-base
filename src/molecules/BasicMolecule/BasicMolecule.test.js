import React from 'react';
import { render } from '@testing-library/react';
import BasicMolecule from './BasicMolecule';

test('BasicMolecule render', () => {
  const { asFragment } = render(
    <BasicMolecule title="example" />,
  );
  expect(asFragment()).toMatchSnapshot();
});
