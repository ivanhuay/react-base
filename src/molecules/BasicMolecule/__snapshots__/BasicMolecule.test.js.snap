// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicMolecule render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <h2>
        example
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
